src/completion-listener.ts

```diff
@@ -86,10 +86,18 @@ export class CompletionListener {
      * @returns A Promise that resolves if the success condition is met, or rejects otherwise.
      */
     listen(commands: Command[]): Promise<CloseEvent[]> {
-        const closeStreams = commands.map((command) => command.close);
+        const closeStreams = commands.map((command) => command.close.pipe());
+
         return Rx.lastValueFrom(
-            Rx.merge(...closeStreams).pipe(
-                bufferCount(closeStreams.length),
+            Rx.combineLatest(closeStreams).pipe(
+                filter((exitInfos) => exitInfos.every((exitInfo) => exitInfo.exitCode !== null)),
+                map((exitInfos) =>
+                    exitInfos.sort((first, second) => {
+                        return (
+                            first.timings.startDate.getTime() - second.timings.startDate.getTime()
```

Comment:
The tests are passing, but they shouldn't.

The definition of conditions `first` and `last` is "the first/last command to exit", but in tests the events are emitted in the tested order, and their start date is the current date. Hence this line didn't break any tests.

In reality though, command 1 might start earlier, but exit later than command 2, so this needs to be

```suggestion
                            first.timings.endDate.getTime() - second.timings.endDate.getTime()
```

Reply:
Got ya, make sense 👍